homeassistant/components/watergate/sensor.py

```diff
@@ -0,0 +1,302 @@
+"""Support for Watergate sensors."""
+
+from collections.abc import Callable, Mapping
+from datetime import UTC, datetime
+import logging
+from typing import Any
+
+from watergate_local_api.models import AutoShutOffReport
+
+from homeassistant.components.sensor import (
+    HomeAssistant,
+    SensorDeviceClass,
+    SensorEntity,
+    SensorStateClass,
+    StateType,
+)
+from homeassistant.const import (
+    SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+    EntityCategory,
+    UnitOfPressure,
+    UnitOfTemperature,
+    UnitOfTime,
+    UnitOfVolume,
+    UnitOfVolumeFlowRate,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import WatergateConfigEntry
+from .coordinator import WatergateAgregatedRequests, WatergateDataCoordinator
+from .entity import WatergateEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+PARALLEL_UPDATES = 0
+
+WIFI_CONNECTION_SENSOR_NAME = "Wifi connection"
```

Comment:
Would be nice it the entity names are translatable.

https://developers.home-assistant.io/docs/internationalization/core#entities